sysboundary/ionosphere.cpp

```diff
@@ -2783,6 +3409,22 @@ namespace SBC {
 } // #pragma omp parallel
 
    }
+   
+   /* Call the heavier operations for DROs to be called only if needed, before an IO.
+    */
+   void SphericalTriGrid::reduceData(
+      FsGrid< fsgrids::technical, FS_STENCIL_WIDTH> & technicalGrid,
+      FsGrid< std::array<Real, fsgrids::bfield::N_BFIELD>, FS_STENCIL_WIDTH> & perBGrid,
+      FsGrid< std::array<Real, fsgrids::dperb::N_DPERB>, FS_STENCIL_WIDTH> & dPerBGrid,
+      dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid
+   ) {
+      if(doTraceOpenClosed) {
+         traceOpenClosedConnection(technicalGrid, perBGrid, dPerBGrid);
+      }
+      if(doTraceFullBox) {
+         traceFullBoxConnection(technicalGrid, perBGrid, dPerBGrid, mpiGrid);
+      }
+   }
```

Comment:
Why can't these instead be called from the corresponding DROs as they are needed?
The ionosphre grid object is accesible from there, too.